lib/lightning_web/channels/attempt_channel.ex

```diff
@@ -177,7 +184,10 @@ defmodule LightningWeb.AttemptChannel do
   end
 
   def handle_in("attempt:log", payload, socket) do
-    Attempts.append_attempt_log(socket.assigns.attempt, payload)
+    %{attempt: attempt, scrubber: scrubber} = socket.assigns
+    payload = Map.update(payload, "message", nil, &maybe_scrub(scrubber, &1))
+
+    Attempts.append_attempt_log(attempt, payload)
```

Comment:
Ergh, I can see why it's awkward here to use the changeset, it's hidden behind `append_attempt_log`. 

I am wary of trying to either scrub anything that _hasn't_ gone through our `LogMessage` casting, I think we should avoiding working on any data structure that isn't what the user will see?

Perhaps we could pass the scrubber pid into `append_attempt_log` as an optional argument, and then add a step to the multi before insert where we can access the changeset?

Or we could, use `LogMessage.cast/1` directly and take it's `{:ok, string()}` result, scrub that and then update payload with that before passing to the Attempts module.

Either way I think we should try and deal with strings here as much as possible when scrubbing. Maybe one day it would make sense to recursively scrub arbitrary structures but I'm not keen on making that change now.

In the future I do see us having a process that collects up log lines in a buffer and inserts them in batches, which would move the theme of scrubbing out of the AttemptChannel and into that process - just an idea of how we could improve this now or later.

Reply:
Done

Reply:
Rad thanks!